tracer/test/Datadog.Trace.ClrProfiler.IntegrationTests/AgentMalfunctionTests.cs

```diff
@@ -45,21 +45,29 @@ public static IEnumerable<object[]> TestData
             => from behaviour in (AgentBehaviour[])Enum.GetValues(typeof(AgentBehaviour))
                from transportType in Transports
                from metadataSchemaVersion in new[] { "v0", "v1" }
-               select new object[] { behaviour, transportType, metadataSchemaVersion };
+               from dataPipelineEnabled in new[] { true, false }
+               select new object[] { behaviour, transportType, metadataSchemaVersion, dataPipelineEnabled };
 
         [SkippableTheory]
         [MemberData(nameof(TestData))]
         [Trait("Category", "EndToEnd")]
         [Trait("RunOnWindows", "True")]
-        public async Task SubmitsTraces(AgentBehaviour behaviour, TestTransports transportType, string metadataSchemaVersion)
+        public async Task SubmitsTraces(AgentBehaviour behaviour, TestTransports transportType, string metadataSchemaVersion, bool dataPipelineEnabled)
         {
             SkipOn.Platform(SkipOn.PlatformValue.MacOs);
             if (transportType == TestTransports.WindowsNamedPipe && !EnvironmentTools.IsWindows())
             {
-                throw new SkipException("Can't use WindowsNamedPipes on non-Windows");
+                throw new SkipException("WindowsNamedPipe transport is only supported on Windows");
             }
 
-            EnvironmentHelper.EnableTransport(transportType);
+            if (transportType == TestTransports.Uds && !EnvironmentTools.IsLinux() && dataPipelineEnabled)
+            {
+                throw new SkipException("Unix Domain Sockets (UDS) transport is only supported
+            }
```

Comment:
This is not true - UDS is supported everywhere for traces 🤔

Reply:
as we talked offline, this is data pipeline issue, and applies to other similar comments.

the issue is https://github.com/tokio-rs/tokio/issues/2201

I have moved the check to TracerSettings, if we detect windows + UDS, disable the data pipeline.